### DISCUSSION TITLE
Injection time and boundary effect

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#635] [Q&A]
With the injector block, how quickly are we allowed to inject the particles while also minimizing boundary effects (e.g I see that if I load them too quickly there is a spike in the E_x field at the max and min x boundary)? Is there a quick rule of thumb? And is there any problem with injecting multiple particles at the same time? 

Practically, what ways can we do to minimize such boundary effects? 

Many thanks,

William.

**Comment by Status-Mirror** [comment]
Hey William,

When many new particles arrive on the boundary, their motion registers a current density which enters Maxwell's equations and a field is established. Injectors are best suited to problems where the injected densities are small compared to the existing plasma. You'll notice that our [injector demos](https://epochpic.github.io/quickstart/basic_examples/injectors.html#) all use low number-densities.

If you have a significant injected density, then it may be better if you intitialise a larger simulation window and pre-load the particles? For example, a window from 0 to 10 microns could be extended to -5 to 10, where the particles you previously injected are now present in the -5 to 0 region. If you move this sheath-field creation to a region which isn't on a simulation boundary (which is already a difficult region to model), then the physics may be more well-behaved.

Would this help your particlular setup?

Cheers,
Stuart

**Reply by cooluser26** [reply]
Thanks for the explanation Stuart. I switched to species initialization instead and the boundary effect is greatly reduced
